File: 02_EstructurasControl/Repetitivas/DoWhile/C#/Ejemplo-02/Ejemplo02.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _2.Dowhile
{
    internal class Program
    {
        static void Main(string[] args)
        {
            /*
             Juego de adivinanza simple: 
            El bucle se repite hasta que el usuario adivina el número secreto,
            garantizando al menos un intento.
             */
            int numeroSecreto = 7; // Número que el usuario debe adivinar
            int intento;
            do
            {
                Console.WriteLine("Adivina el número secreto (entre 1 y 10):");
                intento = Convert.ToInt32(Console.ReadLine());
                if (intento != numeroSecreto)
                {
                    Console.WriteLine("Número incorrecto. Inténtalo de nuevo.");
                }
            } while (intento != numeroSecreto);
            Console.WriteLine("¡Felicidades! Has adivinado el número secreto.");
            Console.ReadKey();

        }
    }
}

File: 02_EstructurasControl/Repetitivas/DoWhile/Python/Ejemplo02.py
# Juego de adivinanza simple: 
# El bucle se repite hasta que el usuario adivina el número secreto,
# garantizando al menos un intento.
numero_secreto = 7  # Número que el usuario debe adivinar (entre 1 y 10)

while True:  # Simula un do-while: ejecuta al menos una vez
    print("Adivina el número secreto (entre 1 y 10):")
    try:
        intento = int(input())  # Convierte la entrada a entero
        if intento != numero_secreto:
            print("Número incorrecto. Inténtalo de nuevo.")
        else:
            break  # Sale del bucle si acierta
    except ValueError:
        print("Entrada no válida. Por favor, ingrese un número entero.")

print("¡Felicidades! Has adivinado el número secreto.")
input("Presione Enter para continuar...")  # Pausa para ver el resultado
